src/parser_json.c

```diff
@@ -452,6 +452,29 @@ json_get_value(struct lyd_node_leaf_list *leaf, struct lyd_node **first_sibling,
         if (data[len++] != '[') {
             LOGVAL(ctx, LYE_XML_INVAL, LY_VLOG_LYD, leaf, "JSON data (expected begin-array)");
             return 0;
+        } else {
+            unsigned int len2 = len;
```

Comment:
doc: I think you need to have a comment here explaining what you are doing.
e.g.
```
// Check for an empty LEAFLIST here.
// This may either be '[ WS ]' or '[ WS none WS ]' (WS = whitespace).
```

Reply:
suggestion: rename `len2` to `empty_list_len` to make it clearer what it is tracking.